verification/src/compiler/version.rs

```diff
@@ -209,4 +265,184 @@ mod tests {
             )
         );
     }
+
+    #[test]
+    fn order_versions() {
+        let ver = check_parsing::<CompilerVersion>;
+
+        // Release only
+        assert!(ver("solc-v0.8.10+commit.fc410830") > ver("solc-v0.8.9+commit.e5eed63a"));
+        // note: version with different hashes shouldn't be equal to implement sorting
+        assert!(ver("solc-v0.8.9+commit.fc410830") > ver("solc-v0.8.9+commit.e5eed63a"));
+        assert!(ver("solc-v0.8.9+commit.e5eed63a") == ver("solc-v0.8.9+commit.e5eed63a"));
+
+        // Nighly only
+        assert!(
+            ver("solc-v0.8.15-nightly.2022.4.4+commit.fd763fa6")
+                > ver("solc-v0.8.2-nightly.2022.3.16+commit.10b581b8")
+        );
+        assert!(
+            ver("solc-v0.8.15-nightly.2022.4.4+commit.fd763fa6")
+                > ver("solc-v0.8.14-nightly.2022.3.16+commit.10b581b8")
+        );
+        assert!(
+            ver("solc-v0.8.14-nightly.2022.4.4+commit.fd763fa6")
+                > ver("solc-v0.8.14-nightly.2022.3.16+commit.10b581b8")
+        );
+        assert!(
+            ver("solc-v0.8.14-nightly.2022.4.4+commit.fd763fa6")
+                > ver("solc-v0.8.14-nightly.2022.4.4+commit.10b581b8")
+        );
+        assert!(
+            ver("solc-v0.8.14-nightly.2022.4.4+commit.fd763fa6")
+                == ver("solc-v0.8.14-nightly.2022.4.4+commit.fd763fa6")
+        );
+
+        // All
+        assert!(
+            ver("solc-v0.5.2+commit.1df8f40c")
+                > ver("solc-v0.5.2-nightly.2018.12.19+commit.88750920")
+        );
+        assert!(
+            ver("solc-v0.5.14+commit.1df8f40c")
+                > ver("solc-v0.5.2-nightly.2018.12.19+commit.88750920")
+        );
+        assert!(
+            ver("solc-v0.5.14-nightly.2019.12.9+commit.d6667560")
+                > ver("solc-v0.5.2+commit.1df8f40c")
+        );
+    }
+
+    fn test_shuffle_and_sort(sorted: Vec<&str>, times: usize) {
+        let sorted_versions: Vec<CompilerVersion> = sorted
+            .clone()
```

Comment:
`.into_iter()` ?

Reply:
I need to use `sorted` after that at line 330, into_iter() takes ownership

Reply:
oh, u are right, will change it to `iter().map()`